tests/unit/misc/test_cmdhistory.py

```diff
@@ -0,0 +1,185 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2015 Alexander Cogneau (acogneau) <alexander.cogneau@gmail.com>:
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Tests for misc.cmdhistory.History."""
+
+import pytest
+
+from qutebrowser.misc import cmdhistory
+
+
+HISTORY = ['first', 'second', 'third', 'fourth', 'fifth']
+
+CONFIG_NOT_PRIVATE = {'general': {'private-browsing': False}}
+CONFIG_PRIVATE = {'general': {'private-browsing': True}}
+
+
+@pytest.fixture
+def hist():
+    return cmdhistory.History(history=HISTORY)
+
+
+def test_no_history():
+    hist = cmdhistory.History()
+    # .history should equal []
+    assert hist.history == []
+
+
+def test_history():
+    hist = cmdhistory.History(history=HISTORY)
+    assert hist.history == HISTORY
+
+
+@pytest.mark.parametrize('tmphist, expected', [(None, False), (HISTORY, True)])
+def test_is_browsing(hist, tmphist, expected):
+    hist._tmphist = tmphist
+    assert hist.is_browsing() == expected
+
+
+def test_start_stop(hist):
+    # We can use is_browsing() because it is tested above
+    assert not hist.is_browsing()
+    hist.start('s')
+    assert hist.is_browsing()
+    hist.stop()
+    assert not hist.is_browsing()
+
+
+def test_start_with_text(hist):
+    """Test start with given 'text'."""
+    hist.start('f')
+    assert 'first' in hist._tmphist
+    assert 'fourth' in hist._tmphist
+    assert 'second' not in hist._tmphist
+
+
+def test_start_no_text(hist):
+    """Test start with no given text."""
+    hist.start('')
+    assert list(hist._tmphist) == HISTORY
+
+
+def test_start_no_items(hist):
+    """Test start with no matching text."""
+    with pytest.raises(cmdhistory.HistoryEmptyError) as excinfo:
+        hist.start('k')
+    assert excinfo is not None
```

Comment:
I'll change this myself, but I wonder what you were trying to do here? `pytest.raises` will abort the test if the expected exception was not raised - so I don't think `excinfo` can ever be `None`.